### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by RostyslavSt**
- notification has it's own status
- creating 'read all' functionality for notifications

### CODE REVIEW

**Review by LanaAvry — APPROVED** [review]

**Review by Kaderimon — DISMISSED** [review]

File: server/routes/devices.js [hunk 1]
```diff
@@ -106,14 +130,37 @@ devicesRouter.route('/:id').put((req, res, next) => {
       device.save()
         .then(device => {
           if (Object.keys(req.body).length  === 1) {
+
+
```
Comment (Kaderimon) [inline server/routes/devices.js]: unnecessary new lines

File: server/routes/devices.js [hunk 2]
```diff
@@ -106,14 +130,37 @@ devicesRouter.route('/:id').put((req, res, next) => {
...
+              userList.forEach((item) => {
+                const objItem = {
+                  userID: item._id,
+                  status: false
+                };
+
+                arr.push(objItem);
+              });
+
+
```
Comment (Kaderimon) [inline server/routes/devices.js]: heere as well

File: src/components/DashboardContent/NotificationsBell.js [hunk 3]
```diff
@@ -46,6 +46,10 @@ class NotificationsBell extends React.Component {
       return { showAllNotify: !prevState.showAllNotify };
     });
   }
+  readAllNotify = () => {
+    this.props.changeStatusNotification();
+    console.log('111');
```
Comment (Kaderimon) [inline src/components/DashboardContent/NotificationsBell.js]: whats that?

File: server/routes/devices.js [hunk 4]
```diff
@@ -75,27 +77,49 @@ devicesRouter.route('/device/:id').get((req, res, next) => {
...
+  Device
+    .findOneAndRemove({ _id: id })
+    .then(device => {
+      let userList = [];
+
+      User.find()
+      .then(users => {
+        userList = [...users];
+      })
+      .then(() => {
```
Comment (Kaderimon) [inline server/routes/devices.js]: whats the point of tht second then() and tab before then is needed

File: src/components/DashboardContent/NotificationsBell.js (hunk 3 above)
Comment (RostyslavSt) [reply src/components/DashboardContent/NotificationsBell.js]: removed

File: server/routes/devices.js (hunk 2 above)
Comment (RostyslavSt) [reply server/routes/devices.js]: removed

File: server/routes/devices.js (hunk 4 above)
Comment (RostyslavSt) [inline server/routes/devices.js]: removed

File: server/routes/devices.js (hunk 1 above)
Comment (RostyslavSt) [reply server/routes/devices.js]: removed

File: server/utils/findAllUsers.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+          userID: item._id,
+          status: false
+        };
+
+
+        arr.push(objItem);
+      });
+      console.log(arr);
+      return arr;
+    });
```
Comment (totoshk) [inline server/utils/findAllUsers.js]: catch error?

**Review by totoshk — APPROVED** [review]

**Pull request merged**